### DISCUSSION TITLE
Key bind is not working on Tabview element

### DISCUSSION THREAD

**Discussion by AndrzejNowakKBA** [TomSchimansky/CustomTkinter#1529] [General]
Hey,
I am facing the problem with key binding when using the tab view in application.
I have a MainWindow file where I am using Tabview element to call other classes.

MainWindow code:
```python
class MainWindow(customtkinter.CTk): 
    def __init__(self):
        self.WindowIsClosed = True
        self.root = customtkinter.CTk()
        self.root.geometry("900x630")
        self.root.protocol("WM_DELETE_WINDOW", self.onExit)
        self.tabControl = customtkinter.CTkTabview(self.root)
        self.tabControl.add("Configuration")
        self.tabControl.add("Labeling")
        self.connectTab = ConnectTab(self.tabControl, self.TCPIP, self.camera, self.NNModel, self)
        self.labelingTab = LabelingTab(self.tabControl, self.NNModel)
```

LabelingTab file:
```python
class LabelingTab:

    def __init__(self,tabControl,NNModel_):

        self.childTab =  tabControl.tab('Labeling')

        self.NNModel = NNModel_


        self.childTab.bind("<Key>", lambda event: checkKeyPressed(event, self, self.NNModel))
```

The function is not executed at all when I just press Labeling tab and tap any key nothing is happening.
When I press Labeling tab, then "Tab" key on keyboard and next I press any button, function is executed.

Is there any better way to make it work ?

**Comment by TomSchimansky** [comment]
Maybe just `bind` to the main window?

**Reply by si1az** [reply]
Worked like a charm 👍

**Comment by Extamble** [comment]
Yeah I did like You said and run the function after it detect specific tab. Thank You.

**Comment by EmbedJunk** [comment]
https://github.com/TomSchimansky/CustomTkinter/discussions/1284
